pccommon/pccommon/tracing.py

```diff
@@ -249,3 +257,49 @@ def _iter_cql(cql: dict, property_name: str) -> Optional[Union[str, List[str]]]:
                             return result
     # No collection was found
     return None
+
+
+def add_stac_attributes_from_search(search_json: str, request: fastapi.Request) -> None:
+    """
+    Try to add the Collection ID and Item ID from a search to the current span.
+    """
+    collection_id, item_id = parse_collection_from_search(
```

Comment:
I think we're effectively running this twice since if it's a /search path we've also called this method from the `trace_request` method. I'm a little hazy about why we didn't want to read the body to get the search from the middleware (from within `trace_request`, really). I think it's because if we attempt to read the body later, we get a "stream consumed" error. I think we should specifically avoid getting the collection/item id from within the `trace_request` method and instead soley get it from the search client like you've got it set up here.

Reply:
This original call is still here. Check this conditional in `trace_request` function, I think we want to remove it now that we are calling _parse_collection_from_search` from within the search client, we can avoid calling it twice

```python
elif path.endswith("/search") or path.endswith("/register"):
     return await _parse_collection_from_search(request)
```